Sources/Swift/Core/Integrations/FramesTracking/SentryScreenFrames.swift

```diff
@@ -0,0 +1,187 @@
+import Foundation
+
+#if (os(iOS) || os(tvOS) || (swift(>=5.9) && os(visionOS)))
+
+/// An array of dictionaries that each contain a start and end timestamp for a rendered frame.
+#if os(iOS)
+@_spi(Private) public typealias SentryFrameInfoTimeSeries = [[String: NSNumber]]
+#endif // os(iOS)
+
+/// Represents screen frame metrics including total, slow, and frozen frames.
+///
+/// - Warning: This feature is not available in `DebugWithoutUIKit` and `ReleaseWithoutUIKit`
+/// configurations even when targeting iOS or tvOS platforms.
+@objc @_spi(Private)
+public final class SentryScreenFrames: NSObject, NSCopying {
+    
+    // MARK: - Properties
+    
+    /// Total number of frames rendered
+    @objc
+    public let total: UInt
+    
+    /// Number of frames that were frozen (took longer than 700ms to render)
+    @objc
+    public let frozen: UInt
+    
+    /// Number of frames that were slow (took longer than 16.67ms but less than 700ms to render)
+    @objc
+    public let slow: UInt
+    
+#if os(iOS)
+    /// Array of dictionaries describing slow frames' timestamps.
+    /// Each dictionary has a start and end timestamp for every such frame,
+    /// keyed under `start_timestamp` and `end_timestamp`.
+    @objc
+    public let slowFrameTimestamps: SentryFrameInfoTimeSeries
+    
+    /// Array of dictionaries describing frozen frames' timestamps.
+    /// Each dictionary has a start and end timestamp for every such frame,
+    /// keyed under `start_timestamp` and `end_timestamp`.
+    @objc
+    public let frozenFrameTimestamps: SentryFrameInfoTimeSeries
+    
+    /// Array of dictionaries describing the screen refresh rate at all points in time that it changes.
+    /// This can happen when modern devices go into low power mode, for example.
+    /// Each dictionary contains keys `timestamp` and `frame_rate`.
+    @objc
+    public let frameRateTimestamps: SentryFrameInfoTimeSeries
+#endif // os(iOS)
+    
+    // MARK: - Initialization
+    
+    /// Creates a `SentryScreenFrames` instance with basic frame metrics.
+    /// - Parameters:
+    ///   - total: Total number of frames rendered
+    ///   - frozen: Number of frozen frames
+    ///   - slow: Number of slow frames
+    @objc public init(total: UInt, frozen: UInt, slow: UInt) {
+#if SENTRY_NO_UIKIT
```

Comment:
`h`: In the Objective-C implementation we checked at the top level if UIKit is available, here we check in the initializer. This changes from a compile-time check to a runtime-check and could introduce problems for users, if they use this class without UIKit